### PR TITLE
Dockerize Socorro end-to-end tests

### PR DESCRIPTION
**Pull request by stephendonner**
@chartjes @davehunt am I close?  Headed in the right direction?

To-do: update docs

Passing ad-hoc job here: https://qa-preprod-master.fxtest.jenkins.stage.mozaws.net/job/socorro.adhoc/87/console

### CODE REVIEW

File: e2e-tests/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM python:2.7-alpine
+WORKDIR /src
+COPY requirements/requirements.txt /src
+RUN pip install pipenv
+RUN pipenv install --system -r requirements.txt --skip-lock --deploy
+COPY . /src
+CMD pytest --driver SauceLabs
```
Comment (davehunt) [inline e2e-tests/Dockerfile]: I would model this on the approach used by Mozillians, where the Docker image has Firefox and GeckoDriver installed, and can run the tests headlessly within the container.

File: e2e-tests/Jenkinsfile [hunk 2]
```diff
@@ -8,60 +8,71 @@ def capabilities = [
...
+      }
+      environment {
+        PYTEST_PROCESSES = "${PYTEST_PROCESSES ?: "auto"}"
+        PULSE = credentials('PULSE')
+        SAUCELABS = credentials('SAUCELABS')
+      }
       steps {
         writeCapabilities(capabilities, 'e2e-tests/capabilities.json')
-        sh "tox -c e2e-tests/tox.ini -e py27"
+        sh "pytest e2e-tests " +
```
Comment (davehunt) [inline e2e-tests/Jenkinsfile]: While this will work, the test path should be the final argument.

File: e2e-tests/setup.cfg [hunk 3]
```diff
@@ -1,4 +1,13 @@
+[flake8]
+ignore = E501
```
Comment (davehunt) [inline e2e-tests/setup.cfg]: We're no longer running flake8 from this location, so we can remove this section.

File: e2e-tests/setup.cfg [hunk 4]
```diff
@@ -1,4 +1,13 @@
...
+
 [isort]
 default_section = THIRDPARTY
 known_first_party = pages
```
Comment (davehunt) [inline e2e-tests/setup.cfg]: As above, we're no longer linting from this context.

**Comment by chartjes** [comment]
In all this I am not seeing where we run the tests using headless Firefox in our own CI infrastructure.

**Comment by davehunt** [comment]
> In all this I am not seeing where we run the tests using headless Firefox in our own CI infrastructure.

We aren't, we're using Sauce Labs so we can run against Windows.

**Comment by chartjes** [comment]
@davehunt These tests are for an admin panel to verify it works, so it doesn't really matter what platform it gets run on. So again I ask the question of why we can't just run them on headless Firefox in our own CI infrastructure. @willkg Is my assumption correct?

**Comment by davehunt** [comment]
> These tests are for an admin panel to verify it works, so it doesn't really matter what platform it gets run on. So again I ask the question of why we can't just run them on headless Firefox in our own CI infrastructure.

See my https://github.com/mozilla-services/socorro/pull/4305#discussion_r163198207 for what's needed to run these headlessly within a Docker container. I have no objection to these running locally and not using Sauce Labs.

File: e2e-tests/Dockerfile (hunk 1 above)
Comment (chartjes) [reply e2e-tests/Dockerfile]: 👍 to that suggestion

**Comment by willkg** [comment]
The tests don't cover the admin at all or anything that requires logging in. They cover a pretty small part of the site. Some navigation, some minor API bits, and some Super Search ui bits.

Here's the list of 24 tests we have:

```
./test_api.py:
   19 :     def test_public_api_navigation(self, base_url):
  121 :     def test_supersearch_fields(self, base_url):
  128 :     def test_platforms(self, base_url):
  135 :     def test_crontabber(self, base_url):
./test_crash_reports.py:
   22 :     def test_that_bugzilla_link_contain_current_site(self, base_url, selenium):
   30 :     def test_that_exploitable_crash_report_is_not_displayed_for_logged_out_users(
   42 :     def test_that_reports_form_has_same_product(self, base_url, selenium, product):
   53 :     def test_that_current_version_selected_in_top_crashers_header(
   68 :     def test_that_top_crasher_filter_all_return_results(self, base_url, selenium):
   79 :     def test_that_top_crasher_filter_browser_return_results(self, base_url, selenium):
   90 :     def test_that_top_crasher_filter_plugin_return_results(self, base_url, selenium):
  102 :     def test_that_top_crashers_reports_links_work(self, base_url, selenium, product):
  115 :     def test_top_crasher_reports_tab_has_uuid_report(self, base_url, selenium):
  140 :     def test_that_7_days_is_selected_default_for_nightlies(self, base_url, selenium):
  148 :     def test_that_only_browser_reports_have_browser_icon(self, base_url, selenium):
  170 :     def test_that_only_plugin_reports_have_plugin_icon(self, base_url, selenium):
  192 :     def test_that_lowest_version_topcrashers_do_not_return_errors(self, base_url, selenium):
./test_search.py:
   13 :     def test_super_search_page_is_loaded(self, base_url, selenium):
   19 :     def test_search_change_facet(self, base_url, selenium):
   37 :     def test_search_change_column(self, base_url, selenium):
   70 :     def test_search_with_one_line(self, base_url, selenium):
   78 :     def test_search_with_multiple_lines(self, base_url, selenium):
   99 :     def test_search_for_valid_signature(self, base_url, selenium):
  108 :     def test_selecting_one_version_doesnt_show_other_versions(self, base_url, selenium):
```

File: e2e-tests/Dockerfile (hunk 1 above)
Comment (stephendonner) [reply e2e-tests/Dockerfile]: @chartjes could that be done in a separate PR, or would you prefer/request that done here, in this one?

**Comment by willkg** [comment]
Tagging with "do not merge" since it seems like this is still a WIP. Let me know when it's done and I can remove that.

**Comment by stephendonner** [comment]
Thanks, @willkg - this is very much a WIP; we've still got to work through (or remove, for now) the geckodriver errors + failures, as well as quite a bit of documentation updates.

File: e2e-tests/Dockerfile (hunk 1 above)
Comment (stephendonner) [reply e2e-tests/Dockerfile]: On IRC, we agreed to leave it out for now, due to crashing Firefox - we'll test/file bugs/re-introduce when it's stable.

**Comment by stephendonner** [comment]
@davehunt HALP!

depositphotos_13984235-stock-illustration-cartoon-man-waving-a-white

**Comment by willkg** [comment]
Thank you!

**Pull request merged**